From 65c5ca53455279b40e6912e2e4b3594097ec04e9 Mon Sep 17 00:00:00 2001
From: "Le.Ma" <Le.Ma@amd.com>
Date: Mon, 18 Dec 2017 14:44:30 +0800
Subject: [PATCH 2967/4131] drm/amdkfd: Fix parameter changes in
 amdgpu_gart_get_vm_pde()

This fixes kfd part affected by patch below:
   - drm/amdgpu: allow get_vm_pde to change flags as well

The modification in this patch is according with kfd branch.

Change-Id: Ie1d76fc5c3d49efca831acf1efb9b17cb59b210e
Signed-off-by: Le.Ma <Le.Ma@amd.com>
Reviewed-by: Xiaojie Yuan <Xiaojie.Yuan@amd.com>
---
 drivers/gpu/drm/amd/amdgpu/amdgpu_amdkfd_gpuvm.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/drivers/gpu/drm/amd/amdgpu/amdgpu_amdkfd_gpuvm.c b/drivers/gpu/drm/amd/amdgpu/amdgpu_amdkfd_gpuvm.c
index 3082f67..236c366 100644
--- a/drivers/gpu/drm/amd/amdgpu/amdgpu_amdkfd_gpuvm.c
+++ b/drivers/gpu/drm/amd/amdgpu/amdgpu_amdkfd_gpuvm.c
@@ -1363,6 +1363,7 @@ static u64 get_vm_pd_gpu_offset(void *vm)
 	struct amdgpu_device *adev =
 		amdgpu_ttm_adev(avm->root.base.bo->tbo.bdev);
 	u64 offset;
+	uint64_t flags = AMDGPU_PTE_VALID;
 
 	BUG_ON(avm == NULL);
 
@@ -1376,7 +1377,7 @@ static u64 get_vm_pd_gpu_offset(void *vm)
 	 * to an actual MC address.
 	 */
 	if (adev->gart.gart_funcs->get_vm_pde)
-		offset = amdgpu_gart_get_vm_pde(adev, offset);
+		amdgpu_gart_get_vm_pde(adev, -1, &offset, &flags);
 
 	return offset;
 }
-- 
2.7.4

